src/DotNetWorker.Grpc/Http/GrpcHttpRequestData.cs

```diff
@@ -31,16 +31,15 @@ public GrpcHttpRequestData(RpcHttp httpData, FunctionContext functionContext)
             _httpData = httpData ?? throw new ArgumentNullException(nameof(httpData));
             _cookies = new Lazy<IReadOnlyCollection<IHttpCookie>>(() =>
             {
-                if (Headers is null)
-                {
-                    return Array.Empty<IHttpCookie>();
-                }
-
-                var cookieString = Headers.FirstOrDefault(item => item.Key.Equals("Cookie", StringComparison.OrdinalIgnoreCase)).Value;
+                var cookieString = Headers
+                    .FirstOrDefault(item => item.Key.Equals("Cookie", StringComparison.OrdinalIgnoreCase))
+                    .Value
+                    ?.ToArray();
 
-                if (cookieString != null && cookieString.Any() && !string.IsNullOrWhiteSpace(cookieString.First()))
+                if (cookieString is [var cookie, ..]
+                    && !string.IsNullOrWhiteSpace(cookie))
                 {
-                    return ToHttpCookies(cookieString.First());
+                    return ToHttpCookies(cookie);
```

Comment:
`ToHttpCookies` is another method which could use `AsSpan` and `Slice` instead of string split multiple times. That should save some allocations.

Reply:
There are a few other areas as well. Will keep those changes coming, but may need to split into multiple PRs.